.github/workflows/checkbox-daily-cross-builds.yaml

```diff
@@ -0,0 +1,158 @@
+  snap-runtime:
+    strategy:
+      fail-fast: false
+      matrix:
+        release: [16, 18, 20, 22, 24]
+        arch: [armhf]
+        include:
+          # other confs built natively in checkbox-daily-native-builds.yaml
+          - release: 16
+            arch: amd64
+          - release: 16
+            arch: arm64
+    runs-on: ${{ matrix.release == 16 && 'ubuntu-20.04' || 'ubuntu-latest' }}
+    timeout-minutes: 1200 #20h, this will timeout sooner due to inner timeouts
+    name: Runtime ${{ matrix.release }} (${{ matrix.arch }})
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          persist-credentials: false
+
+      - name: Copy over the common files for series ${{ matrix.release }}
+        run: |
+          cd checkbox-core-snap/
+          sudo apt update && sudo apt install -qq -y python3-setuptools-scm
+          ./prepare.sh series${{ matrix.release }}
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+
+      - name: Build (retries on fail)
+        uses: Wandalen/wretry.action@master
+        with:
+          attempt_limit: 5
+          action: canonical/snapcraft-multiarch-action@v1
+          with: |
+            architecture: ${{ matrix.arch }}
+            path: checkbox-core-snap/series${{ matrix.release }}
+
+      - uses: actions/upload-artifact@v4
+        name: Upload logs on failure
+        if: failure()
+        with:
+          name: runtime-build-log-${{ matrix.release }}-${{ matrix.arch }}
+          path: |
+            /home/runner/.cache/snapcraft/log/
+            /home/runner/.local/state/snapcraft/log/
+            checkbox-core-snap/series${{ matrix.release }}/checkbox*.txt
+
+      - uses: actions/upload-artifact@v4
+        name: Upload the snap as artifact
+        with:
+          name: runtime_snap${{ matrix.release }}_${{ matrix.arch }}.snap
+          path: checkbox-core-snap/*/*.snap
+
+      - name: Publish track
+        if: ${{ github.event_name == 'workflow_call' }}
+        uses: canonical/action-publish@v1
+        env:
+          SNAPCRAFT_STORE_CREDENTIALS: ${{ secrets.SNAPCRAFT7_CREDS }}
+        with:
+          snap: checkbox-core-snap/*/*.snap
+          release: latest/edge
+
+  snap-frontend:
+    strategy:
+      fail-fast: false
+      matrix:
+        type: [classic, uc]
+        release: [16, 18, 20, 22, 24]
+        arch: [armhf, amd64, arm64]
+        exclude:
+          # other confs built natively in checkbox-daily-native-builds.yaml
+          - release: 18
+            arch: amd64
+          - release: 18
+            arch: arm64
+          - release: 20
+            arch: amd64
+          - release: 20
+            arch: arm64
+          - release: 22
+            arch: amd64
+          - release: 22
+            arch: arm64
+          - release: 24
+            arch: amd64
+          - release: 24
+            arch: arm64
+    runs-on: ${{ matrix.release == 16 && 'ubuntu-20.04' || 'ubuntu-latest' }}
```

Comment:
Can you add a comment of why UC16 requires 20.04 to be built? For our future selves :)